### PR TITLE
[Scala3] Provide correct dialect to `Mtags.allToplevels` calls.

### LINKED ISSUE #2891: Issue parsing Scala 3 and macros
**Describe the bug**

I'm not 100% sure what's going on as I didn't dive too deep into this, but wanted to capture what @keynmol mentioned on Gitter here.

Either there is an issue with Scalameta parsing this or we aren't correctly identifying that the source we are in is Scala 3. However, I do see that in general it recognizes that the build target is Scala 3, so I don't think that's the issue. Basically when you open a specific file with a macro in it you get a giant error and in the logs you see something along the lines of:

```
SEVERE: Internal error: /Users/ckipp/Documents/scala-workspace/expecty/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala:32: error: unclosed character literal
    '{
    ^
java.util.concurrent.CompletionException: /Users/ckipp/Documents/scala-workspace/expecty/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala:32: error: unclosed character literal
    '{
    ^
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
        at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:40)
        at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:38)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: /Users/ckipp/Documents/scala-workspace/expecty/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala:32: error: unclosed character literal
    '{
    ^
        at scala.meta.internal.tokenizers.Reporter.syntaxError(Reporter.scala:23)
        at scala.meta.internal.tokenizers.Reporter.syntaxError$(Reporter.scala:23)
        at scala.meta.internal.tokenizers.Reporter$$anon$1.syntaxError(Reporter.scala:33)
        at scala.meta.internal.tokenizers.Reporter.syntaxError(Reporter.scala:25)
        at scala.meta.internal.tokenizers.Reporter.syntaxError$(Reporter.scala:25)
        at scala.meta.internal.tokenizers.Reporter$$anon$1.syntaxError(Reporter.scala:33)
        at scala.meta.internal.tokenizers.LegacyScanner.fetchSingleQuote$1(LegacyScanner.scala:469)
        at scala.meta.internal.tokenizers.LegacyScanner.fetchToken(LegacyScanner.scala:474)
        at scala.meta.internal.tokenizers.LegacyScanner.nextToken(LegacyScanner.scala:251)
        at scala.meta.internal.mtags.ScalaToplevelMtags.acceptBalancedDelimeters(ScalaToplevelMtags.scala:274)
        at scala.meta.internal.mtags.ScalaToplevelMtags.loop(ScalaToplevelMtags.scala:172)
        at scala.meta.internal.mtags.ScalaToplevelMtags.indexRoot(ScalaToplevelMtags.scala:50)
        at scala.meta.internal.mtags.MtagsIndexer.index(MtagsIndexer.scala:19)
        at scala.meta.internal.mtags.MtagsIndexer.index$(MtagsIndexer.scala:18)
        at scala.meta.internal.mtags.ScalaToplevelMtags.index(ScalaToplevelMtags.scala:40)
        at scala.meta.internal.mtags.Mtags$.allToplevels(Mtags.scala:96)
        at scala.meta.internal.metals.DefinitionProvider.fromMtags(DefinitionProvider.scala:212)
        at scala.meta.internal.metals.DefinitionProvider.$anonfun$positionOccurrence$3(DefinitionProvider.scala:174)
        at scala.Option.orElse(Option.scala:447)
        at scala.meta.internal.metals.DefinitionProvider.$anonfun$positionOccurrence$1(DefinitionProvider.scala:174)
        at scala.Option.flatMap(Option.scala:271)
        at scala.meta.internal.metals.DefinitionProvider.positionOccurrence(DefinitionProvider.scala:169)
        at scala.meta.internal.metals.DocumentHighlightProvider.$anonfun$documentHighlight$1(DocumentHighlightProvider.scala:33)
        at scala.collection.immutable.List.map(List.scala:293)
        at scala.meta.internal.metals.DocumentHighlightProvider.documentHighlight(DocumentHighlightProvider.scala:29)
        at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$documentHighlights$1(MetalsLanguageServer.scala:1335)
        at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
        at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        ... 4 more
```

This happens with both Bloop and sbt BSP.

**To Reproduce**

1. Clone https://github.com/eed3si9n/expecty
2. Open `expecty/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala` and import project
3. See errors

**Expected behavior**

No errors in sight

**Installation:**

- Operating system: macOS
- Editor: NeoVim
- Metals version: v 0.10.4+87-7b502251-SNAPSHOT
--

**Search terms**

Scala 3 macros

### PR DESCRIPTION
**Pull request by dos65**
Fixes #2891

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala [hunk 1]
```diff
@@ -370,21 +363,6 @@ final class BuildTargets(
       .map(_.getId())
   }
 
-  private def unsafeInferBuildTarget(
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala:373]: This method is no longer needed as now we write source-jar-name in the dependencies directory.

File: metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala [hunk 2]
```diff
@@ -267,7 +268,10 @@ class MetalsTreeViewProvider(
   ): Option[TreeViewNodeRevealResult] = {
     val input = path.toInput
     val occurrences =
-      Mtags.allToplevels(input).occurrences.filterNot(_.symbol.isPackage)
+      Mtags
+        .allToplevels(input, dialects.Scala213)
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala:272]: Hardcoded 213 here. TreeViewProvider doesn't works with scala3 - #2859

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala:272]: Could you add a comment above referencing the issue?

**Review by tgodzik — APPROVED** [review]
LGTM aside from a very minor comment.

**Review by ckipp01 — APPROVED** [review]
Nice! LGTM!

**Pull request merged**